src/main/kotlin/com/example/ecs/components/WelcomeMessage.kt

```diff
@@ -1,4 +1,4 @@
-package com.example
+package com.example.ecs.components
```

Comment:
Im not a big fan of this split.

Splitting by type instead of concern seems like smelly organization.

Reply:
Wait, this is a package thing not source root. I really don't mind here.